@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -572,17 +572,40 @@ export class VisibleTileSet {
         const worldGeoPoint = this.options.projection.unprojectPoint(worldCenter);
         const result: TileKeyEntry[] = [];
         const startOffset = Math.round(worldGeoPoint.longitude / 360.0);
-        const worldLengthHorizontal =
-            Math.tan(MathUtils.degToRad(this.camera.fov) * this.camera.aspect) *
-            -this.camera.position.z;
+        const cameraPitch = MapViewUtils.extractYawPitchRoll(this.camera.quaternion).pitch;
+        // Compute the number of tiles to show, this is based on the camera FOV, the aspect ratio
+        // and the tilt (because for example, if the user has rotated 90 degrees, and the camera is
+        // tilted, then we want to show tiles going to the horizon (but obviously not infinity)). To
+        // calculate this distance, we form a right angle triangle, the angle [[totalAngle]] is at
+        // the camera, it is half of the FOV and the camera tilt. The hypotenuse is formed from the
+        // field of view and the intersection with the ground plane, and the adjacent distance is
+        // simply the camera z height.
+        const totalAngleRad = MathUtils.degToRad(this.camera.fov / 2) + cameraPitch;
+        // Ensure that the aspect is >= 1.
+        const aspect = this.camera.aspect > 1 ? this.camera.aspect : 1 / this.camera.aspect;
+        const worldLengthHorizontalFull = Math.tan(totalAngleRad * aspect) * this.camera.position.z;
+        const worldLengthHorizontalSmallerHalf =
+            Math.tan(cameraPitch * aspect) * this.camera.position.z;
+        // We subtract the smaller half, to get the larger half. This represents the world distance
+        // from the look at point to the intersection of the camera left frustum with the plane.s
+        const worldLengthHorizontal = worldLengthHorizontalFull - worldLengthHorizontalSmallerHalf;
         const worldLeftPoint = new THREE.Vector3(
             worldCenter.x - worldLengthHorizontal,
             worldCenter.y,
             worldCenter.z
         );
         const worldLeftGeoPoint = this.options.projection.unprojectPoint(worldLeftPoint);
-        const offsetRange = Math.ceil(
-            Math.abs((worldGeoPoint.longitude - worldLeftGeoPoint.longitude) / 360)
+        // We multiply by Sqrt(2) because we need to account for a rotated view (in which case there
+        // are more tiles that can be seen).
+        const offsetRange = MathUtils.clamp(
+            Math.ceil(
+                Math.abs((worldGeoPoint.longitude - worldLeftGeoPoint.longitude) / 360) * 1.414
```

Comment:
1.414 looks quite like a magic number... In C++, we have `M_SQRT2`, would it make sense to have something like a `MathUtils.Sqrt2` constant? Or, if we don't need it anywhere else, a `const Sqrt2 = 1.414;` ?

Reply:
I did a search for 1.414 and didn't find it used elsewhere so i have added a local const.